src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1871,6 +1989,19 @@ internal static uint GetValEscape(BoundExpression expr, uint scopeOfTheContainin
             {
                 case BoundKind.DefaultExpression:
                 case BoundKind.Parameter:
+                case BoundKind.ThisReference:
+                    // always returnable
+                    return Binder.ExternalScope;
+
+                case BoundKind.MakeRefOperator:
+                case BoundKind.RefValueOperator:
+                    // for compat reasons
```

Comment:
Technically result of MakeRefOperator cannot be returned, but that is enforced by not allowing methods to have TypedReference as a return type.  
If we assign val-escape be "top-level  here we will break compat on scenarios like

```cs
TypedReference tr;    // uninitialized and thus "returnable"
tr = __makeref(someVariable);   // do not want an error here. This is an existing scenario.
```